### ISSUE TITLE
add changelog

### ISSUE THREAD

**Issue by leela52452** [nproth/pin#16]
can you please add changelog.
what is [CHANGELOG](https://en.m.wikipedia.org/wiki/Changelog) ?

**Comment by nproth** [comment]
I will try to be more conclusive about the changes made in different files in the git log.
However, I think that the CHANGELOG file which - to be honest - is not up to date should only include user facing changes like new features or bug fixes without the information what was changed in the source code.
But I can see that some code changes are probably hard to follow and I will try to better document the code and any changes made.
